infrastructure/async/src/main/java/tech/pegasys/teku/infrastructure/async/LimitedThrottlingTaskQueue.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Consensys Software Inc., 2025
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.infrastructure.async;
+
+import java.util.concurrent.RejectedExecutionException;
+import java.util.function.Supplier;
+import org.hyperledger.besu.plugin.services.MetricsSystem;
+import tech.pegasys.teku.infrastructure.metrics.TekuMetricCategory;
+
+public class LimitedThrottlingTaskQueue implements TaskQueue {
```

Comment:
ok so we had throttling, now we have throttling and limited.... they're both semantically very similar in english so i think we'd want to understand why use limitedThrottling over throttling etc - the names really dont help us understand when to use one over the other.

Reply:
any suggestion? :)